source/extensions/filters/listener/proxy_protocol/proxy_protocol.h

```diff
@@ -38,9 +40,11 @@ class Config {
 
 typedef std::shared_ptr<Config> ConfigSharedPtr;
 
+enum ProxyProtocolVersion { kUnknown = -1, kInProgress = -2, kV1 = 1, kV2 = 2 };
+
 /**
- * Implementation the PROXY Protocol V1 listener filter
- * (http://www.haproxy.org/download/1.5/doc/proxy-protocol.txt)
+ * Implementation the PROXY Protocol listener filter
```

Comment:
Please specify at a high level which parts of the specification are supported in this implementation.